### DISCUSSION TITLE
Any reference available for math behind different Instructions for TP.

### DISCUSSION THREAD

**Discussion by moradza** [e3nn/e3nn#355] [General]
Hi, 

I have problem grasping the instructions part of TP e.g., 'uvu', ''uvuv', 'uvw', or ... . Is there any reference explaining these instructions. 

Thanks,

**Comment by mariogeiger** [comment]
This code is quite easy to read I think: https://github.com/e3nn/e3nn-jax/blob/main/e3nn_jax/_src/legacy/core_tensor_product.py#L516-L568

Tell me if you need clarifications

UPDATE: fixed broken link

**Reply by tkella47** [reply]
Can this be updated? Link is broken

**Reply by mitkotak** [reply]
https://github.com/e3nn/e3nn-jax/blob/8c502c1046f5121a1c77cf71eccfa1dbbc71ef5d/e3nn_jax/_src/legacy/core_tensor_product.py#L487-L523
